src/coreclr/jit/lsraxarch.cpp

```diff
@@ -3213,18 +3221,47 @@ int LinearScan::BuildMul(GenTree* tree)
         return BuildSimple(tree);
     }
 
-    // ToDo-APX : imul currently doesn't have rex2 support. So, cannot  use R16-R31.
-    int              srcCount      = BuildBinaryUses(tree->AsOp());
+    bool isUnsignedMultiply    = tree->IsUnsigned();
+    bool requiresOverflowCheck = tree->gtOverflowEx();
+    bool useMulx               = tree->OperGet() != GT_MUL && isUnsignedMultiply &&
+                   compiler->compOpportunisticallyDependsOn(InstructionSet_BMI2);
+
+    // ToDo-APX : imul currently doesn't have rex2 support. So, cannot use R16-R31.
```

Comment:
I am not sure how to handle the APX thing so I left the comment, it will now need to be handled for both instructions.

mulx seems at least to support evex encoding so if apx and bmi2 is supported it should probably be no issue